File: FaucetPayCSharp (Unofficial) .NetCore/HTTP/HttpClientRequester.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Threading.Tasks;
using Models;
using Config;

namespace Http
{
    /// <summary>
    /// *** MADE FOR GITHUB ***
    /// WRITTEN BY ROMULO MEIRELLES.
    /// UPWORK: https://www.upwork.com/freelancers/~01fcbc5039ac5766b4
    /// CONTACT WHATSAPP: https://wa.me/message/KWIS3BYO6K24N1
    /// CONTACT TELEGRAM: https://t.me/Romulo_Meirelles
    /// GITHUB: https://github.com/Romulo-Meirelles
    /// DISCORD: đąяķňέs§¢øďε#2786
    /// </summary>
    /// <remarks></remarks>

    public partial class HttpClientRequester : RequesterBase, IDisposable
    {

        private readonly HttpClient _Client;
        private bool _DisposeClient;
        public HttpClientRequester(ApiConfig Configuration, HttpClient Client) : base(Configuration)
        {
            _Client = Client;
        }
       public override async Task<T> PostAsync<T>(string resource, Dictionary<string, string> parameters = null,bool noThrow = false)
        {
            if (resource == "faucetlist")
                        _Client.BaseAddress = new Uri(Configuration.BaseFaucetList);
            var result = await CreateRequestAsync(resource, HttpMethod.Post, parameters).ConfigureAwait(false);
            var response = await result.Content.ReadAsStringAsync().ConfigureAwait(false);
            var deserialized = Deserialize<T>(response);
            HandleError(deserialized.Status, deserialized.Message, noThrow: noThrow);
            return deserialized;
        }

        public static void ConfigureClient(HttpClient Client, ApiConfig Configuration)
        {
            Client.BaseAddress = new Uri(Configuration.BaseUrl);
        }
        protected internal virtual Task<HttpResponseMessage> CreateRequestAsync(string Resource, HttpMethod Method, Dictionary<string, string> Parameters)
        {
            var content = new FormUrlEncodedContent(CreateParameterPairs(Parameters));
            return _Client.SendAsync(new HttpRequestMessage(Method, new Uri(Resource, UriKind.Relative)) { Content = content });
        }
        public static HttpClientRequester Create(ApiConfig Config, HttpClient Client = null, bool DisposeClient = false)
        {
            if (Client is null)
                DisposeClient = true;
            Client = Client ?? new HttpClient();
            ConfigureClient(Client, Config);
            return new HttpClientRequester(Config, Client) { _DisposeClient = DisposeClient };
        }
        public void Dispose()
        {
            if (_DisposeClient)
                _Client.Dispose();
        }
    }
}

File: FaucetPay (Unofficial) .NetCore/HTTP/HttpClientRequester.vb
Imports System.Net.Http
Imports FaucetPay.Models
Imports FaucetPay.Config

Namespace Http
    ''' <summary>
    ''' *** MADE FOR GITHUB ***
    ''' WRITTEN BY ROMULO MEIRELLES.
    ''' UPWORK: https://www.upwork.com/freelancers/~01fcbc5039ac5766b4
    ''' CONTACT WHATSAPP: https://wa.me/message/KWIS3BYO6K24N1
    ''' CONTACT TELEGRAM: https://t.me/Romulo_Meirelles
    ''' GITHUB: https://github.com/Romulo-Meirelles
    ''' DISCORD: đąяķňέs§¢øďε#2786
    ''' </summary>
    ''' <remarks></remarks>

    Public Class HttpClientRequester
        Inherits RequesterBase
        Implements IDisposable

        Private ReadOnly _Client As HttpClient
        Private _DisposeClient As Boolean
        Public Sub New(ByVal Configuration As ApiConfig, ByVal Client As HttpClient)
            MyBase.New(Configuration)
            _Client = Client
        End Sub
        Public Overrides Async Function PostAsync(Of T As FaucetPayResponse)(Resource As String, Optional Parameters As Dictionary(Of String, String) = Nothing, Optional noThrow As Boolean = False) As Task(Of T)
            If Resource = "faucetlist" Then _Client.BaseAddress = New Uri(Configuration.BaseFaucetList)
            Dim Result = Await CreateRequestAsync(Resource, HttpMethod.Post, Parameters).ConfigureAwait(False)
            Dim Response = Await Result.Content.ReadAsStringAsync().ConfigureAwait(False)
            Dim Deserialized = Deserialize(Of T)(Response)
            HandleError(Deserialized.Status, Deserialized.Message, noThrow:=noThrow)
            Return Deserialized
        End Function
        Public Shared Sub ConfigureClient(Client As HttpClient, ByVal Configuration As ApiConfig)
            Client.BaseAddress = New Uri(Configuration.BaseUrl)
        End Sub
        Protected Friend Overridable Function CreateRequestAsync(ByVal Resource As String, ByVal Method As HttpMethod, ByVal Parameters As Dictionary(Of String, String)) As Task(Of HttpResponseMessage)
            Dim content = New FormUrlEncodedContent(CreateParameterPairs(Parameters))
            Return _Client.SendAsync(New HttpRequestMessage(Method, New Uri(Resource, UriKind.Relative)) With {.Content = content})
        End Function
        Public Shared Function Create(ByVal Config As ApiConfig, ByVal Optional Client As HttpClient = Nothing, ByVal Optional DisposeClient As Boolean = False) As HttpClientRequester
            If Client Is Nothing Then DisposeClient = True
            Client = If(Client, New HttpClient())
            HttpClientRequester.ConfigureClient(Client, Config)
            Return New HttpClientRequester(Config, Client) With {._DisposeClient = DisposeClient}
        End Function
        Public Sub Dispose() Implements IDisposable.Dispose
            If _DisposeClient Then _Client.Dispose()
        End Sub
    End Class
End Namespace
